frontend/src/components/UploadPhotoDialog.vue

```diff
@@ -72,14 +72,31 @@ export default Vue.extend({
             this.uploadProgress = 0;
             this.uploadTotal = this.photos.length;
 
-            const results = await Promise.all(this.photos.map(async photoFile => {
-                const photoBytes = new Uint8Array(await photoFile.arrayBuffer());
-                const result = await createPhoto(this.albumId, photoBytes);
+            let results = [];
+            for(const photoFile of this.photos) {
+                while(true) {
+                    const photoBytes = new Uint8Array(await photoFile.arrayBuffer());
+                    const result: boolean | TooManyRequests | undefined = await createPhoto(this.albumId, photoBytes);
+
+                    if(result === true) {
+                        break;
+                    }
+
+                    if(result instanceof TooManyRequests) {
+                        console.log("Got HTTP 429. Waiting 1s");
+                        await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
Set a retry-after header in the response when returning a 429 and use that to determine the timeout instead of waiting an arbitrary 1 second

Reply:
Fair yeah.

AWS doesn't return the 429 unfortunately, it just does 403. But I'll cook something up

Reply:
Is there no way to estimate a rate limit or obtain one from AWS? If there isn't any way I am also fine with you approach